docs/plugin_developers_guide/07_rules_plugins.rst

```diff
@@ -0,0 +1,432 @@
+.. Rules Plugins
+
+.. |br| raw:: html
+
+   <br/>
+
+.. Links
+.. |average| raw:: html
+
+   <a href="../plugins/fledge-rule-average/index.html">Moving Average Rule plugin</a>
+
+.. |code| raw:: html
+
+   <a href="https://github.com/fledge-iot/fledge-rule-average.git">Moving Average Rule source code</a>
+
+
+Notification Rule Plugins
+=========================
+
+Notification rule  plugins are used by the notification system to
+evaluate a rule and trigger a notification based on the result of
+that evaluation. They are the logic that allows the event to be
+notified to that other system or device.
+
+
+Notification delivery plugins may be written in C or C++ and have a very
+simple interface. The plugin mechanism and a subset of the API is common
+between all types of plugins including notification rules.  This documentation
+is based on the |code|. The |average| calculates a moving average of the values
+sent to the notification instance and will trigger the notification to be sent
+when the value of a datapoint varies from the calculated average by more than
+a specified percentage.
+
+Configuration
+-------------
+
+Notification Rule  plugins use the same configuration mechanism as the rest of
+Fledge, using a JSON document to describe the configuration parameters. As
+with any other plugin the structure is defined by the plugin and retrieve
+by the *plugin_info* entry point. This is then matched with the database
+content to pass the configured values to the *plugin_init* entry point.
```

Comment:
> As with any other plugin

Similar to other plugins,

> This is then matched with the database

Subsequently, this is aligned with the database content